### DISCUSSION TITLE
Peripheral registry viewer plugin doesn't show anything

### DISCUSSION THREAD

**Discussion by ksrp1984** [eclipse-embed-cdt/eclipse-plugins#612] [Debug Plug-ins Q&A] [answered]
I run debugging via OpienOSD:

Config options: 
    -f /board/stm32f0discovery.cfg

Commands:
    set mem inaccessible-by-default off
    set arch arm
    set remotetimeout 250

SVD file:
    STM32F0x8.svd

Debbuger:
    ST-LINK

Peripheral registry viewer plugin doesn't show anything!
Is there any way to find out if Eclipse is trying to load the svd file? Even if I specify the wrong path to the file, I do not receive any errors.

OS:
    Ubuntu 24.04 LTS

Eclispe:
     Version: 2024-06 (4.32.0)
      Build id: 20240606-1231

OpenOCD Plugin:
      Embedded C/C++ OpenOCD Debugging	6.6.0.202404031712	org.eclipse.embedcdt.debug.gdbjtag.openocd.feature.group	Eclipse Embedded CDT

**Comment by TommyMurphyTM1234** [comment]
Maybe check the `workspace/.metadata/.log` file for clues?

**Reply by ksrp1984** [reply]
!ENTRY org.eclipse.embedcdt.packs.core 4 1 2024-07-20 15:26:29.809
!MESSAGE /mnt/files/distributiv/Linux/xpack-openocd-0.12.0-3/bin/openocd -c gdb_port 3333 -c telnet_port 4444 -c tcl_port 6666 -f /board/stm32f0discovery.cfg -c echo "Started by GNU MCU Eclipse"

!ENTRY org.eclipse.embedcdt.packs.core 4 1 2024-07-20 15:26:29.835
!MESSAGE /mnt/files/distributiv/Linux/arm-gnu-toolchain-13.3.rel1-x86_64-arm-none-eabi/bin/arm-none-eabi-gdb --interpreter=mi2 --nx

**Comment by ksrp1984** [comment]
Ubuntu no longer supports arm-none-eabi-gdb. The repository contains only gdb-multiarch. There are also no more libncursesw.so.5 and libtinfo.so.5 libraries that arm-none-eabi-gdb requires. I created links to get arm-none-eabi-gdb working:

sudo ln -s /usr/lib/x86_64-linux-gnu/libncursesw.so.6.4 /usr/lib/x86_64-linux-gnu/libncursesw.so.5
sudo ln -s /usr/lib/x86_64-linux-gnu/libtinfo.so.6.4 /usr/lib/x86_64-linux-gnu/libtinfo.so.5

Perhaps this could be the reason.

log for gdb-multiarch (works with cortex but peripheral registry viewer plugin doesn't show anything):

!ENTRY org.eclipse.embedcdt.managedbuild.cross.riscv.core 4 1 2024-07-20 15:49:18.287
!MESSAGE /mnt/files/distributiv/Linux/xpack-openocd-0.12.0-3/bin/openocd -c gdb_port 3333 -c telnet_port 4444 -c tcl_port 6666 -f /board/stm32f0discovery.cfg -c echo "Started by GNU MCU Eclipse"

!ENTRY org.eclipse.embedcdt.managedbuild.cross.riscv.core 4 1 2024-07-20 15:49:18.324
!MESSAGE gdb-multiarch --interpreter=mi2 --nx

**Reply by ilg-ul** [reply]
> Ubuntu no longer supports arm-none-eabi-gdb. The repository contains only gdb-multiarch. There are also no more libncursesw.so.5 and libtinfo.so.5 libraries that arm-none-eabi-gdb requires.

There are good reasons why I created the xPack binary distributions, and one of them is that all dependent libraries are distributed in the package, so the binaries should run on any system.

**Reply by ilg-ul** [reply]
> Is there any way to find out if Eclipse is trying to load the svd file? Even if I specify the wrong path to the file, I do not receive any errors.

Check the consoles, in one of them there should be messages related to the packs.

**Comment by ksrp1984** [comment]
Debugger console:

GNU gdb (xPack GNU Arm Embedded GCC x86_64) 13.2.90.20231008-git
Copyright (C) 2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "--host=x86_64-pc-linux-gnu --target=arm-none-eabi".

**Comment by ksrp1984** [comment]
Log:

!ENTRY org.eclipse.embedcdt.packs.core 4 1 2024-07-20 23:33:14.963
!MESSAGE /mnt/files/distributiv/Linux/xpack-openocd-0.12.0-3/bin/openocd -c gdb_port 3333 -c telnet_port 4444 -c tcl_port 6666 -f /board/stm32f0discovery.cfg -c echo "Started by GNU MCU Eclipse"

!ENTRY org.eclipse.embedcdt.packs.core 4 1 2024-07-20 23:33:15.005
!MESSAGE /mnt/files/distributiv/Linux/xpack-arm-none-eabi-gcc-13.2.1-1.1/bin/arm-none-eabi-gdb --interpreter=mi2 --nx

**Comment by ksrp1984** [comment]
registry viewer plugin doesn't show anything

**Comment by TommyMurphyTM1234** [comment]
I think that @ilg-ul means run Eclipse from a terminal shell console and then see what, if any, log/error messages you see there when you try to use the peripheral viewer.

**Comment by ksrp1984** [comment]
No errors.
Only:
2024 11:38:50 PM org.apache.aries.spifly.BaseActivator log
INFO: Registered provider org.slf4j.simple.SimpleServiceProvider of service org.slf4j.spi.SLF4JServiceProvider in bundle slf4j.simple

**Comment by TommyMurphyTM1234** [comment]
What about with `-debug -consoleLog`?

- https://help.eclipse.org/latest/index.jsp?topic=/org.eclipse.platform.doc.isv/reference/misc/runtime-options.html

**Comment by ksrp1984** [comment]
[ConsoleLog.txt](https://github.com/user-attachments/files/16322302/ConsoleLog.txt)

**Comment by ilg-ul** [comment]
I think @ksrp1984 is right, Eclipse does not log anything related to the SVD file, at least not when everything is fine.

The file is not used by gdb or by the debugger, it is only used by the Eclipse debug plug-in.

If the peripheral viewer does not show anything, the first suspect is the SVD file, which might not be valid.

Where did you get the SVD file from?

**Comment by ksrp1984** [comment]
svd file from ST (Cube)

**Comment by TommyMurphyTM1234** [comment]
Maybe try validating the SVD file using the arm tools? 

- https://github.com/eclipse-embed-cdt/eclipse-plugins/issues/610#issuecomment-2217430503

**Comment by ilg-ul** [comment]
I created a new M0 project using the xpack template, I added the `/Users/ilg/Library/CMSIS-Packs/Keil/STM32F0xx_DFP/2.1.0/CMSIS/SVD/STM32F0x8.svd` path to the debug configuration, and here is the result:

![Screenshot 2024-07-21 at 00 05 58](https://github.com/user-attachments/assets/782c6446-af1c-4050-a68e-5bae244b6c5a)

**Reply by ilg-ul** [reply, marked as answer]
I agree that my SVD file is a bit older, so I don't know how well behaves the one you use.

Please try again with the same file I used and is known to work:

[STM32F0x8.svd.zip](https://github.com/user-attachments/files/16322370/STM32F0x8.svd.zip)

**Comment by ksrp1984** [comment]
Thank you! It works!
